pulsar-broker-common/src/main/java/org/apache/pulsar/broker/authentication/utils/AuthTokenUtils.java

```diff
@@ -90,39 +92,18 @@ public static String createToken(Key signingKey, String subject, Optional<Date>
     }
 
     public static byte[] readKeyFromUrl(String keyConfUrl) throws IOException {
-        if (keyConfUrl.startsWith("data:")) {
-            return readDataUrl(keyConfUrl);
+        if (keyConfUrl.startsWith("data:") || keyConfUrl.startsWith("file:")) {
+            try {
+                return ByteStreams.toByteArray((InputStream) new URL(keyConfUrl).getContent());
+            } catch (Exception e) {
+                throw new IOException(e);
+            }
         } else if (keyConfUrl.startsWith("env:")) {
```

Comment:
Couldn't you have a URL handler for this also?

Reply:
Yes, adding

Reply:
Actually, there's one tricky part. 

The key, passed as an env variable, needs to be base64 encoded. That might not be needed in general case for `env:` (eg: a token would be fine as it is). 

It's kind of confusing and we'd have to define something similar to `env:base64,MY_SECRET_KEY_VAR`

Reply:
I think we shoudn’t define encoding field on env scheme to keep it simple. The requirement comes from application use case (which is this plugin here), so the application should take care of it.
If you get URLConnection from the URL object, you can check the scheme name as protocol name.

Reply:
I didn’t read key generation part, but if the key is in PEM format we can use the same format for all scheme and decode the raw data read from the InputStream in a consistent way, I think.

Reply:
I have removed the `env:` schema as it was not working properly. Java seems to be mangling long var values. We can add later when we can make it work.